### PR TITLE
[Escha] Import portal functionality. Enable all portals

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Imports logic from the escha branch, to add portal functionality.
- Adds all portal scripts and enables warping.

About Ethereal Droplet: Out of scope for this PR. Will implement in the future, along the npcs that give them, once I get captures.

## Steps to test these changes

Go to Escha Zi'tah `!zone 288` or Escha Ru'Aun `!zone 289` or Reisenjima `!zone 291`
Travel arround examining portals. Note portals getting unlocked properly.

### CODE REVIEW

File: scripts/globals/teleport_systems/eschan_portals.lua [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        portalsUnlocked = portalsUnlocked + 1
+        activatedMask   = player:getTeleport(xi.teleport.type.ESCHAN_PORTAL, maskOffset)
+    end
+
+    -- Check if we have other portals to warp to. Do not display menu if not.
+    if portalsUnlocked <= 1 then
+        activatedMask = 1
+    end
+
+    player:startEvent(9100, 0, activatedMask, zoneID, portalNumber, unlockBit, player:getCurrency("escha_silt"), getPortalCost(player), 0)
```
Comment (WinterSolstice8) [inline scripts/globals/teleport_systems/eschan_portals.lua]: Is this what CI wants you to do? looks odd to me?

Comment (Xaver-DaRed) [reply scripts/globals/teleport_systems/eschan_portals.lua]: I had this like `9100,0` with no space in between, so CI didn't like it. As it shouldn't, btw XD

File: tools/migrations/035_escha_portals_unlock.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import array
+import mysql.connector
+
+def migration_name():
+    return "Adding eschan_portals and vorseals columns to char_unlocks table"
```
Comment (WinterSolstice8) [inline tools/migrations/035_escha_portals_unlock.py]: This says "and vorseals" but there's only one new column, are they combined or is this something leftover from the branch?

Comment (Xaver-DaRed) [reply tools/migrations/035_escha_portals_unlock.py]: oops, leftover. Originaly, In the Escha branch, vorseal logic was included. I decided to separate both.

Will fix now

**Review by zach2good — APPROVED** [review]

**Review by claywar — APPROVED** [review]

**Pull request merged**